src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -3683,25 +3632,22 @@ bool BlockchainLMDB::for_all_transactions(std::function<bool(const crypto::hash&
       break;
     if (ret)
       throw0(DB_ERROR(lmdb_error("Failed to enumerate transactions: ", ret).c_str()));
-    transaction tx;
+    blobdata owned_txblob;
+    blobdata_ref bdref;
     if (pruned)
     {
-      blobdata_ref bd{reinterpret_cast<char*>(v.mv_data), v.mv_size};
-      if (!parse_and_validate_tx_base_from_blob(bd, tx))
```

Comment:
Question, what do you think about keeping `parse_and_validate_tx_base_from_blob`  in `#ifdef DEBUG` preprocessor? Although we don't use debug macro in `db_lmdb.cpp`. 

I realize that you want to remove transaction-specific code from db_lmdb,  but at the same time, it looks like a valuable check, particularly when debugging.

Reply:
Check the other `BlockchainDB::for_all_transactions` method. We do still have the `for_all_transactions()` method, but now the parsing is done outside of this class.